### PR TITLE
1.5/Configuration of .travis.yml to build successfully tests: 

### PR DESCRIPTION
**Pull request by albertsh**
- edit of addons_path_list in test_server.py
 - add some dependencies
 - add of test for account_split_invoice module

### CODE REVIEW

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+from datetime import datetime
+from openerp import netsvc
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class test_split_invoice(common.TransactionCase):
+
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Remove this empty line

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    """ Test Project split invoice """
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Put a better comment : list the test scenarios, without giving the details.

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    def setUp(self):
+        super(test_split_invoice, self).setUp()
+        self.invoice_id = self._create_invoice('SAJ/2015/0002')
+        self.invoice_line_id1 = self._create_invoice_line('service 1', '70.0')
+        self.invoice_line_id2 = self._create_invoice_line('service 2', '80.0')
+
+    def _create_invoice(self, invoice_name):
+
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Remove empty line.

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        # Set the update_posted to True to make them cancelable
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: to make <b>invoice</b> cancelable.

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        journal_obj = self.registry('account.journal')
+        journal_obj.write(self.cr, self.uid, 1, {
+            'update_posted': True,
+        })
+
+        partner = self.registry('res.partner')
+        partner_id = partner.create(self.cr, self.uid, {
+            'name': 'Kevin',
+
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Remove empty line

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+            wizard.invoice_id.account_id, original_account)
+        self.assertEqual(
+            wizard.invoice_id.journal_id, original_journal)
+        self.assertEqual(
+            original_amount,
+            wizard.invoice_id.amount_total + new_invoice.amount_total
+        )
+
+    def test_open_invoice(self):
+
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Remove empty line

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        # test for invoice in 'open' state
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Use docstrings (google for it) for method comments. Explain in this comment the test scenario (split an open invoice).

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+        wf_service = netsvc.LocalService('workflow')
+        wf_service.trg_validate(
+            self.uid, 'account.invoice', self.invoice_id,
+            'invoice_open', self.cr)
+
+        self._test_write_lines(invoice)
+
+    def test_draft_invoice(self):
+
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Remove empty line and update method comment like above.

**Comment by ecino** [comment]
Please put public methods before your private methods inside your tests.

File: account_invoice_split_invoice/tests/__init__.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emanuel Cino <ecino@compassion.ch>
```
Comment (ecino) [inline account_invoice_split_invoice/tests/__init__.py]: You can put your name instead.

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 10]
```diff
@@ -0,0 +1,120 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emanuel Cino <ecino@compassion.ch>
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: You can put your name instead.

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 11]
```diff
@@ -0,0 +1,120 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: 2015

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+        self.invoice_line_id1 = self._create_invoice_line('service 1', '70.0')
+        self.invoice_line_id2 = self._create_invoice_line('service 2', '80.0')
+
+    def _create_invoice(self, invoice_name):
+        """ Set the update_posted to True to make invoice cancelable """
+        journal_obj = self.registry('account.journal')
+        journal_obj.write(self.cr, self.uid, 1, {
+            'update_posted': True,
+        })
+        partner = self.registry('res.partner')
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: partner_obj

**Comment by ecino** [comment]
In your test, move the private methods (starting with _xxx) after the public methods.

**Comment by ecino** [comment]
For the tests of split_invoice, I would do the two tests (draft and open) for two different invoices to be sure that one test does not interfer with the other one.

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+                (1, self.invoice_line_id1, {'split': True})], "")
+        invoice_obj = self.registry('account.invoice')
+        new_invoice = invoice_obj.browse(self.cr, self.uid, new_invoice_id)
+
+        # Test if the lines have been exactly copied
+        self.assertEqual(wizard.invoice_id.id, self.invoice_id)
+        self.assertNotEqual(wizard.invoice_id.id, new_invoice_id)
+        self.assertEqual(wizard.invoice_id.name, original_name)
+        self.assertEqual(
+            wizard.invoice_id.partner_id, original_partner)
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: You should also do this assert on the new invoice

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+        self.assertEqual(
+            wizard.invoice_id.date_invoice, original_invoice_date)
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: You should also do this assert on the new invoice

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+        self.assertEqual(
+            wizard.invoice_id.account_id, original_account)
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: You should also do this assert on the new invoice

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+            'price_subtotal': amount,
+            'invoice_id': self.invoice_id,
+        })
+        return invoice_line_id
+
+    def _test_write_lines(self, invoice):
+        # Test if the invoice has been copied
+        original_amount = invoice.amount_total
+        original_name = invoice.name
+        original_partner = invoice.partner_id
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: You should rather take only the partner_id.
```original_partner_id = invoice.partner_id.id```

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 17]
```diff
@@ -0,0 +1,120 @@
...
+        original_invoice_date = invoice.date_invoice
+        original_account = invoice.account_id
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Read only the account_id (invoice.account_id.id)

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+        original_journal = invoice.journal_id
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Read only journal_id

File: account_invoice_split_invoice/tests/test_split_invoice.py [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+        self.assertNotEqual(wizard.invoice_id.id, new_invoice_id)
+        self.assertEqual(wizard.invoice_id.name, original_name)
+        self.assertEqual(
+            wizard.invoice_id.partner_id, original_partner)
+        self.assertEqual(
+            wizard.invoice_id.date_invoice, original_invoice_date)
+        self.assertEqual(
+            wizard.invoice_id.account_id, original_account)
+        self.assertEqual(
+            wizard.invoice_id.journal_id, original_journal)
```
Comment (ecino) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: Test on the journal_id.
You should also do this assert on the new invoice

File: recurring_contract/__init__.py [hunk 20]
```diff
@@ -1,9 +1,9 @@
 # -*- encoding: utf-8 -*-
 ##############################################################################
 #
-#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
+#    Copyright (C) 2015 Compassion CH (http://www.compassion.ch)
 #    Releasing children from poverty in Jesus' name
-#    @author: Cyril Sester <csester@compassion.ch>
+#    @author: Emanuel Cino <ecino@compassion.ch>
```
Comment (ecino) [inline recurring_contract/__init__.py]: This change is weird. You can put back Cyril Sester as the author.

File: recurring_contract/model/recurring_contract.py [hunk 21]
```diff
@@ -180,7 +191,9 @@ def _get_contract_from_group(group_obj, cr, uid, group_ids, context=None):
             digits_compute=dp.get_precision('Account'),
             store={
                 'recurring.contract': (lambda self, cr, uid, ids, c=dict():
-                                       ids, ['contract_line_ids'], 20),
+                                       ids, ['contract_line_ids'], 40),
+                'recurring.contract.line': (_get_contract_from_line,
+                                            ['amount'], 30),
```
Comment (ecino) [inline recurring_contract/model/recurring_contract.py]: Add field `quantity` as well for trigerring the computation.

File: recurring_contract/model/recurring_invoicer.py [hunk 22]
```diff
@@ -22,6 +22,7 @@
 
 
 class recurring_invoicer(orm.Model):
+
```
Comment (ecino) [inline recurring_contract/model/recurring_invoicer.py]: Remove empty line

File: recurring_contract/tests/__init__.py [hunk 23]
```diff
@@ -0,0 +1,18 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emanuel Cino <ecino@compassion.ch>
```
Comment (ecino) [inline recurring_contract/tests/__init__.py]: Put your name

File: recurring_contract/tests/test_recurring_contract.py [hunk 24]
```diff
@@ -0,0 +1,189 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
```
Comment (ecino) [inline recurring_contract/tests/test_recurring_contract.py]: 2015

File: recurring_contract/tests/test_recurring_contract.py [hunk 25]
```diff
@@ -0,0 +1,189 @@
...
+from openerp.tests import common
+from datetime import datetime
+from openerp.tools import DEFAULT_SERVER_DATE_FORMAT as DF
+from openerp import netsvc
+import logging
+logger = logging.getLogger(__name__)
+
+
+class test_recurring_contract(common.TransactionCase):
+
```
Comment (ecino) [inline recurring_contract/tests/test_recurring_contract.py]: Remove empty line

File: recurring_contract/tests/test_recurring_contract.py [hunk 26]
```diff
@@ -0,0 +1,189 @@
...
+        property_account_receivable = self.registry('account.account').search(
+            self.cr, self.uid, [
+                ('type', '=', 'receivable'),
+                ('user_type', '=', account_type)
+            ])[0]
+        property_account_payable = self.registry('account.account').search(
+            self.cr, self.uid, [('type', '=', 'payable')])[0]
+
+        # Creation of a partner
+        partner = self.registry('res.partner')
```
Comment (ecino) [inline recurring_contract/tests/test_recurring_contract.py]: partner_obj

File: recurring_contract/tests/test_recurring_contract.py [hunk 27]
```diff
@@ -0,0 +1,189 @@
...
+        partner_id = partner.create(self.cr, self.uid, {
+            'name': 'Monsieur Client 137',
+            'property_account_receivable': property_account_receivable,
+            'property_account_payable': property_account_payable,
+        })
+        # Creation of payement terms
+        payment_term = self.registry('account.payment.term')
```
Comment (ecino) [inline recurring_contract/tests/test_recurring_contract.py]: payment_term_obj

File: recurring_contract/tests/test_recurring_contract.py [hunk 28]
```diff
@@ -0,0 +1,189 @@
...
+            datetime.today().strftime(DF))
+        self.contract_line_id = self._create_contract_line(self.contract_id,
+                                                           '40.0')
+
+    def _create_contract(self, start_date, group_id, next_invoice_date):
+        """
+            Create a contract. For that purpose we have created a partner
+            to get his id
+        """
+        # Creation of an account
```
Comment (ecino) [inline recurring_contract/tests/test_recurring_contract.py]: Lines 76-96 are copy-paste from lines 36-54. You can remove them and use the partner_id you created in SetUp().

**Comment by ecino** [comment]
Having two files for the two tests of module recurring_contract makes a lot of useless similar code (copy-paste). You can merge the two files and have the two tests in two separate methods inside the same file so that you can make your SetUp only once, and create two contracts using the same methods for doing your two tests.

File: recurring_contract/tests/test_recurring_contract.py [hunk 29]
```diff
@@ -0,0 +1,189 @@
...
+        contract_act = contract.browse(self.cr, self.uid, self.contract_id)
+        self.assertEqual(contract_act.state, 'active')
+        invoicer = self.registry('recurring.invoicer')
+        invoicer_id = contract_act.button_generate_invoices()
+        invoicer_obj = invoicer.browse(self.cr, self.uid, invoicer_id)
+        invoices = invoicer_obj.invoice_ids
+        nb_invoice = len(invoices)
+        # 2 invoices must be generated with our
+        self.assertEqual(nb_invoice, 2)
+        # parameters
```
Comment (ecino) [inline recurring_contract/tests/test_recurring_contract.py]: Why is your comment break with the line of code ? Put it in a single line.

**Comment by ecino** [comment]
You can increase the coverage by adding the following scenarios to your test :

- instead of calling directly the method generate_invoices from contract_group, try creating a recurring_invoicer_wizard and generate invoices from the wizard
- unlink partner that has active contract should raise an error
- unlink active contract should raise an error
- change invoice lines of active contract and see if invoices are well updated
- copy of a contract (why not for the second test copy first contract to make a second one ?)
- make several contracts in a same group to generate invoice with several invoice lines and then cancel one contract, see if invoices are still correct. See if cancel invoices remain.
- test cancel_invoices method of recurring_invoicer.

=> It should increase coverage of recurring_contract.py, recurring_invoicer.py, recurring_invoicer_wizard.py

File: account_invoice_split_invoice/tests/__init__.py (hunk 9 above)
Comment (albertsh) [inline account_invoice_split_invoice/tests/__init__.py]: done

File: account_invoice_split_invoice/tests/test_split_invoice.py (hunk 10 above)
Comment (albertsh) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: done

File: account_invoice_split_invoice/tests/test_split_invoice.py (hunk 11 above)
Comment (albertsh) [inline account_invoice_split_invoice/tests/test_split_invoice.py]: done

**Comment by ecino** [comment]
:+1:

**Pull request merged**